File: explorer-backend/src/Models/API/AddressResponse.cs
using ExplorerBackend.Models.System;

namespace ExplorerBackend.Models.API;

public class AddressResponse
{
    public bool Fetched { get; set; }
    public bool IsValid { get; set; }
    public ValidateAddress? Address { get; set; }
    public bool AmountFetched { get; set; }
    public double Amount { get; set; }
    public int? Version { get; set; }
    public string? Hash { get; set; }
    public string? ScriptHash { get; set; }
}

File: explorer-frontend/models/API/AddressResponse.ts
export interface ValidateAddressResult {
    isvalid: boolean,
    address: string | null,
    scriptPubKey: string | null,

    isscript: boolean | null,
    iswitness: boolean | null,

    witness_version: number | null,
    witness_program: string | null,

    isextkey: boolean | null,

    isstealthaddress: boolean | null,
    prefix_num_bits: number | null,
    prefix_bitfield: string | null
}

export interface AddressResponse {
    fetched: boolean,
    address: ValidateAddressResult | null,
    amountFetched: boolean,
    amount: number,
    version: number | null,
    hash: string | null
}